darts/utils/data/tabularization.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import pandas as pd
+
+from darts.logging import raise_if
+from darts.timeseries import TimeSeries
+
+
+def create_lagged_data(
```

Comment:
I think we should probably keep this function private (unless we have a reason not to?)